lambdas/SaveSnippetToS3/index.py

```diff
@@ -0,0 +1,75 @@
+import json
+import urllib
+import re
+import os
+from datetime import datetime
+from boto3.s3.transfer import ClientError
+import string
+import boto3 # AWS SDK for Python
+
+s3 = boto3.client('s3', 'us-west-2')
+client = boto3.client('lambda')
+
+#Verifies existance of index file; if absent, creates index file. Writes file to S3.
+def updateIndexFile(snippetKey, bucket, snippetData, user):
+    try:
+        indexData = s3.get_object(Bucket=bucket, Key=user + '/index.json')['Body'].read()
+    except ClientError as error:
+        indexData = '{}'
+
+    index = json.loads(indexData)
+    index[snippetKey] = snippetData
+    s3.put_object(Body=json.dumps(index), Bucket=bucket, Key=user +'/index.json')
+
+#Saves the given body to the appropiate bucket under the assigned key.
+def saveToS3(bucket, key, body):
+    try:
+        s3.put_object(Body=body, Bucket=bucket, Key=key)
+    except ClientError as error:
+        err_code = str(error.response['Error']['Code'])
+        print 'Error putting object %s into bucket %s. Make sure they exist and ' \
+              'your bucket is in the same region as this function.' % (key, bucket)
+        raise error
+
+def lambda_handler(event, context):
+    #Extract needed data from the event
+    accessToken = event['headers']['Authorization']
+    userID = event['pathParameters']['user_id']
+
+    authorizeTokenName = os.environ['authorizeTokenName']
+
+    #Invoke the auth. lambda to verify the accessToken mathches the userID for the requested resource
+    lambda_auth = client.invoke(
+        FunctionName=os.environ['authorizeTokenName'],
+        Payload=json.dumps({'accessToken': accessToken, 'userID': userID}))
+
+    lambda_payload_resp = json.loads(lambda_auth['Payload'].read())
+
+    if(lambda_payload_resp['statusCode'] == '400'):
```

Comment:
Oh, come on with your "As a python developer..."  Parens after an if mean the same in every language, even if you don't *have* to have them